userspace/libsinsp/filter/parser.cpp

```diff
@@ -28,7 +28,7 @@ limitations under the License.
 // these follow the POSIX standard
 #define RGX_NOTBLANK            "(not[[:space:]]+)"
 #define RGX_IDENTIFIER          "([a-zA-Z]+[a-zA-Z0-9_]*)"
-#define RGX_FIELDNAME           "([a-zA-Z]+[a-zA-Z0-9_]*(\\.[a-zA-Z]+[a-zA-Z0-9_]*)+)"
+#define RGX_FIELDNAME           "([a-zA-Z]+[a-zA-Z0-9_\\[\\]]*(\\.[a-zA-Z]+[a-zA-Z0-9_]*)+)"
```

Comment:
> I'm currently looking at reworking the implementation to be able to index directly off of the class name (e.g. `fd[2].name`) and implement this in the `sinsp_filter_check` class as an optional feature for any child filter class. It's a little more effort, but I think it will be a cleaner approach and allow the feature to be more easily used by other filters. Let me know what you think.


Hey @allanembedded 

Supporting arguments in the field class (i.e. `class[*].field`) is not part of our convention and may break many codebase assumptions. It would also drastically change the condition syntax. IMO, it's something that can be introduced only after an extensive discussion. I don't think you should do that in this PR. 

cc @jasondellaluce WDYT?

Reply:
> I don't think you should do that in this PR.

OK, so what would be acceptable in this PR? The underlying functionality seems to work as is, so for me it's just how you'd like the syntax to look, clearly this is something you'd be better placed to suggest than I. I have no strong feelings on this, my main concern is supporting the functionality.

Reply:
Btw, I believe the functionality is ok, and I like it. 
Now, we have to decide:
- the syntax 
- the naming

For the former, I guess  _class_._field_[*]._subfield_ could work. But I'd like to get a second opinion from @jasondellaluce (he worked a lot on the condition syntax, so he may provide valuable feedback)

If the above syntax is ok, we can then discuss the naming. The first thing that came to my mind is `proc.fds[*].type` :thinking: but not sure if it's ok.
@falcosecurity/libs-maintainers @falcosecurity/rules-maintainers  WDYT?

Reply:
Any thoughts on this @jasondellaluce?

Reply:
After thinking again about this, I realized that even `proc.fds[*].type` is not ok. To be consistent with other fields, the syntax should be  `class.field[]` as initially proposed by @loresuso :point_down: 

> One observation I have is that we should find another name for this, all the fields are in the form of `class.field` and this should follow this rule as well like `class.field[]`. Can we think of something better?

For example, `fdlist.names[]` works in this way :point_down:  
https://falco.org/docs/reference/rules/supported-fields/#field-class-process

So, at this point, I believe we should use something like `fds.names[]`, `fds.types[]`, etc...

Reply:
I think extending the condition syntax should be our last resort in general and I don't think it's worth it for this use case. It is already a bit fragile, and I'd advise on updating the grammar only if we have compelling improvements in mind for the language in general.

This being said, I think what you're trying to achieve here can be accomplished without altering the current language by either:
- Fields like `fd.type` to optionally-accept an argument, such as `fd.type[1]`. Without an argument it will return info for the file descriptor of the current event, whereas with an argument it will allow the selection based on the fd number.
- Add new "plural" fields such as `fd.types` that optionally accepts an argument, such as `fd.types[1]`. Without argument it will return a list of infos (that you can work on with operators like `in`, `intersects`, etc...), whereas with an argument it will allow the selection based on the fd number.

Of the two, I would prefer going for option two. Overall, not only this would avoid changing the language, but it will be more predictable and manageable at compile time too. WDYT?

Reply:
> Add new "plural" fields such as fd.types that optionally accepts an argument, such as fd.types[1]. Without argument it will return a list of infos (that you can work on with operators like in, intersects, etc...), whereas with an argument it will allow the selection based on the fd number.

Agreed this should still cover the use case in question. Will give it a shot and update the PR if it works out.